doc/reference/dune-project/package.rst

```diff
@@ -26,6 +26,29 @@ is used when generating OPAM files (see :doc:`generate_opam_files`).
 
       Package dependencies, as :token:`~pkg-dep:dep_specification`.
 
+   .. describe:: (documentation <doc-spec>)
+
+      The documentation field allows to specify documentation options for the
+      package.
+
+      It may contain the following fields:
+
+      .. describe:: (url <url>)
+
+         Where the documentation is hosted. Overrides the corresponding global
+         field (see :doc:`documentation`).
+
+      .. describe:: (depends <dep-specification>)
+
+         Lists the packages whose docs is referenced in an ``mli`` or ``mld``
+         file. For instance, to be able to reference ``cmdliner``'s tutorial
+         with ``{!/cmdliner/tutorial}``, ``cmdliner`` needs to be added here.
+
+         The syntax for this field is :token:`~pkg-dep:dep_specification`.
+
+      This field can also be the same as the corresponding global field (see
+      :doc:`documentation`), in which case it takes precedence.
+
```

Comment:
Also it needs to retain the `versionadded:: 2.0` from the original documentation.